### ISSUE TITLE
Cannot load saved ginza model by spacy.load()

### ISSUE THREAD

**Issue by hi-ra-y** [megagonlabs/ginza#132]
#### my environment
Ubuntu 18.04.3 LTS
python 3.6.9

#### an extract of pip list
ginza                         4.0.1
spacy                         2.3.2    
srsly                         1.0.2 
SudachiDict-core              20200722
SudachiPy                     0.4.9 

#### my code
```python
import spacy
nlp=spacy.load('ja_ginza')
nlp.to_disk('./save_dir')
nlp2=spacy.load('./save_dir')
```
We usually implement code to do nlp tasks between `nlp=spacy.load('ja_ginza')` and `nlp.to_disk('./save_dir')`.
(Like nlp.add_pipe(), nlp.begin_trainning(), nlp.update()…）
But I want to reproduce the error in more easy situation now, and found this simple code causes the same one.

#### Errors
ValueError                                Traceback (most recent call last)
<ipython-input-9-da9ff2e19e36> in <module>()
----> 1 nlp2=spacy.load('./save_dir')

10 frames
/usr/local/lib/python3.6/dist-packages/spacy/__init__.py in load(name, **overrides)
     28     if depr_path not in (True, False, None):
     29         warnings.warn(Warnings.W001.format(path=depr_path), DeprecationWarning)
---> 30     return util.load_model(name, **overrides)
     31 
     32 

/usr/local/lib/python3.6/dist-packages/spacy/util.py in load_model(name, **overrides)
    170             return load_model_from_package(name, **overrides)
    171         if Path(name).exists():  # path to model data directory
--> 172             return load_model_from_path(Path(name), **overrides)
    173     elif hasattr(name, "exists"):  # Path or Path-like to model data
    174         return load_model_from_path(name, **overrides)

/usr/local/lib/python3.6/dist-packages/spacy/util.py in load_model_from_path(model_path, meta, **overrides)
    220             component = nlp.create_pipe(factory, config=config)
    221             nlp.add_pipe(component, name=name)
--> 222     return nlp.from_disk(model_path, exclude=disable)
    223 
    224 

/usr/local/lib/python3.6/dist-packages/spacy/language.py in from_disk(self, path, exclude, disable)
    972             # Convert to list here in case exclude is (default) tuple
    973             exclude = list(exclude) + ["vocab"]
--> 974         util.from_disk(path, deserializers, exclude)
    975         self._path = path
    976         return self

/usr/local/lib/python3.6/dist-packages/spacy/util.py in from_disk(path, readers, exclude)
    688         # Split to support file names like meta.json
    689         if key.split(".")[0] not in exclude:
--> 690             reader(path / key)
    691     return path
    692 

/usr/local/lib/python3.6/dist-packages/spacy/language.py in <lambda>(p, proc)
    967                 continue
    968             deserializers[name] = lambda p, proc=proc: proc.from_disk(
--> 969                 p, exclude=["vocab"]
    970             )
    971         if not (path / "vocab").exists() and "vocab" not in exclude:

/usr/local/lib/python3.6/dist-packages/ginza/compound_splitter.py in from_disk(self, path, **_kwargs)
    198             )
    199         )
--> 200         util.from_disk(path, serializers, [])

/usr/local/lib/python3.6/dist-packages/spacy/util.py in from_disk(path, readers, exclude)
    688         # Split to support file names like meta.json
    689         if key.split(".")[0] not in exclude:
--> 690             reader(path / key)
    691     return path
    692 

/usr/local/lib/python3.6/dist-packages/ginza/compound_splitter.py in <lambda>(p)
    195         serializers = OrderedDict(
    196             (
--> 197                 ("cfg", lambda p: self._set_config(srsly.read_json(p))),
    198             )
    199         )

/usr/local/lib/python3.6/dist-packages/srsly/_json_api.py in read_json(location)
     48         data = sys.stdin.read()
     49         return ujson.loads(data)
---> 50     file_path = force_path(location)
     51     with file_path.open("r", encoding="utf8") as f:
     52         return ujson.load(f)

/usr/local/lib/python3.6/dist-packages/srsly/util.py in force_path(location, require_exists)
     19         location = Path(location)
     20     if require_exists and not location.exists():
---> 21         raise ValueError("Can't read file: {}".format(location))
     22     return location
     23 

ValueError: Can't read file: save_dir/CompoundSplitter/cfg

#### about saved model
 - Checking 'save_dir' after doing above simple code, acutually there is no `CompoundSplitter` directory.
 The components is here.
```
$find ./save_dir 
./save_dir
./save_dir/tokenizer
./save_dir/tokenizer/cfg
./save_dir/tagger
./save_dir/tagger/model
./save_dir/tagger/cfg
./save_dir/tagger/tag_map
./save_dir/vocab
./save_dir/vocab/lookups.bin
./save_dir/vocab/key2row
./save_dir/vocab/vectors
./save_dir/vocab/strings.json
./save_dir/meta.json
./save_dir/ner
./save_dir/ner/model
./save_dir/ner/moves
./save_dir/ner/cfg
./save_dir/parser
./save_dir/parser/model
./save_dir/parser/moves
./save_dir/parser/cfg
```

- If I duplicate the `tokenizer` directory and rename it to `CompoundSplitter`, then I can execute `spacy.load('./save_dir')`.
 
----- 
So, I think I do something wrong to save model (or to load model), but I couldn`t find the exact way.
Could you please teach me some sample code?

**Comment by hi-ra-y** [comment]
I'm sorry if it's strange English…　I'm japanese.

**Comment by hiroshi-matsuda-rit** [comment]
Thank you for reporting. I found the reason why serialized directory does not contain the directories for custom components. We need to add `@component` annotations to the classes of custom components of GiNZA. I've started to implement it. @hi-ra-y

**Comment by hiroshi-matsuda-rit** [comment]
I'll release this fix as v4.0.2 along with other features.

**Comment by hi-ra-y** [comment]
I see. Thank you for your quick response.
